src/test/java/org/superbiz/ColorResourceTest.java

```diff
@@ -68,57 +75,67 @@ public static WebArchive createDeployment() {
      *
      * This allows the test itself to be agnostic of server information or even
      * the name of the webapp
-     *
      */
-    @ArquillianResource
-    private URL webappUrl;
-
+    private @ArquillianResource URL webappUrl;
 
     @Test
     public void postAndGet() throws Exception {
+        final WebTarget resourceTarget = buildResourceTarget();
 
-        // POST
+        // GET
         {
-            final WebClient webClient = WebClient.create(webappUrl.toURI());
-            final Response response = webClient.path("color/green").post(null);
+            final Response response = resourceTarget.path("color").request().get();
+            assertEquals(200, response.getStatus());
+            final String content = slurp((InputStream) response.getEntity());
+            assertEquals("white", content);
+        }
 
+        // POST
+        {
+            final Response response = resourceTarget.path("color/green").request().post(null);
             assertEquals(204, response.getStatus());
         }
 
         // GET
         {
-            final WebClient webClient = WebClient.create(webappUrl.toURI());
-            final Response response = webClient.path("color").get();
-
+            final Response response = resourceTarget.path("color").request().get();
             assertEquals(200, response.getStatus());
-
             final String content = slurp((InputStream) response.getEntity());
-
             assertEquals("green", content);
         }
-
     }
 
     @Test
     public void getColorObject() throws Exception {
-
-        final WebClient webClient = WebClient.create(webappUrl.toURI());
-        webClient.accept(MediaType.APPLICATION_JSON);
-
-        final Color color = webClient.path("color/object").get(Color.class);
-
+        final WebTarget resourceTarget = buildResourceTarget();
+        final Response response = resourceTarget.path("color/object").request(MediaType.APPLICATION_JSON).get();
+        final Color color = response.readEntity(Color.class);
         assertNotNull(color);
         assertEquals("orange", color.getName());
         assertEquals(0xE7, color.getR());
         assertEquals(0x71, color.getG());
         assertEquals(0x00, color.getB());
     }
 
+    private WebTarget buildResourceTarget() throws Exception {
+        final ClientBuilder client = ClientBuilder.newBuilder();
+        try {
+            // client side
+            client.register(Class.forName(JSON_PROVIDER_CLASS));
```

Comment:
I stubbed in support in the pom for switching to the remote adapter, so I made the test portable. I didn't reference the JohnzonProvider directly so that the class would compile without any adapters or extra jars on the classpath.